File: Algorithm/Recursion/Solution/N과 M(2) - boj/BOJ15650.py
def printSequence(bucket):
    if(len(bucket) - 1 == M):
        print(' '.join(map(str, bucket[1:])))
        return

    for i in range(bucket[-1] + 1, N + 1):
        bucket.append(i)
        printSequence(bucket)
        bucket.pop()


N, M = map(int, input().split())
printSequence([0])

File: Algorithm/Recursion/Solution/N과 M(2) - boj/BOJ15650.c
#include <stdio.h>

int N, M;

void printSequence(int cnt, int last, int bucket[])
{
    if (cnt == M)
    {
        for (int i = 0; i < M; ++i)
        {
            printf("%d ", bucket[i]);
        }

        printf("\n");

        return;
    }

    for (int i = last + 1; i <= N; ++i)
    {
        bucket[cnt] = i;
        printSequence(cnt + 1, i, bucket);
    }
}

int main(void)
{
    scanf("%d %d", &N, &M);
    int bucket[M];
    printSequence(0, 0, bucket);
    return 0;
}